rust/core-lib/src/view.rs

```diff
@@ -900,16 +909,12 @@ impl View {
         width_cache: &mut WidthCache,
         drift: InsertDrift,
     ) {
-        self.lines.after_edit(text, delta, width_cache, client);
-        if !self.lines.has_soft_breaks() {
-            let (iv, new_len) = delta.summary();
-            let start = self.line_of_offset(last_text, iv.start());
-            let end = self.line_of_offset(last_text, iv.end()) + 1;
-            let new_end = self.line_of_offset(text, iv.start() + new_len) + 1;
-            self.lc_shadow.edit(start, end, new_end - start);
-        } else {
-            //TODO (really) actual minimal invalidation, based on what wrapped.
-            self.set_dirty(text);
+        let visible = self.first_line..self.first_line + self.height;
```

Comment:
Personally I would find spaces around the `..` clearer, as it binds more loosely than the `+`.

Reply:
Looks like there's another case like this a few lines earlier. This sounds like a rustfmt change though, so fixing it would likely affect other parts of the code. Adding parentheses around the `+` term might help here though.

Reply:
yea my impression as well was that rustfmt wouldn't let me, bu it seems to be fine. I don't have much of a preference either way?

Reply:
Reformatting this would definitely make it a good bit easier to scan. Other than that I'd say this is good to go.

Reply:
Yeah we'll want to make sure the topmost visible line stays where it is in the view. That should be quite doable once we get line numbers to updates? Additionally I think rewrapping should prioritize the visible chunk a lot more, in sqlite.c I see quite a lot of nothing when I resize and scroll.

But yes, it's a great improvement and I was assuming this to be another checkpoint like the previous stuff :)

Reply:
it's a bit tricker than just line numbers, because the line cache operates on visible lines; when a wrap happens earlier in the document we need to essentially send the client the delta in the number of earlier lines, so that it can manually adjust the current position.

Reply:
And can you clarify what you mean by "quite a lot of nothing"? are you being left in an empty region?

Reply:
Yes, basically everything except the old visible chunk is empty when resizing & scrolling in quick succession.

Hmm do you mean the cache operates on logical (wrapped) lines? That would indeed make it a bit tricky.

Reply:
In my usage logical lines = 'real' lines; delineated by 0, EOF, or newline boundaries. Visual lines = 'wrapped' lines, which may not be delineated in this way.

And yes, the cache works on visual (wrapped) lines; otherwise the client would need to know where to insert the breaks.

"Empty" means that it is not rendered? Is this on a debug build? (That shouldn't happen)

Reply:
Ah, so it does show me the region where it *thinks* I should be right after resizing but that moves around the document while staying "static" in the view, hence empty regions when scrolling. Also it seems like I can't really interact with the text field while it rewraps? (select text etc) Hard to say for sure since it keeps warping around though.

Reply:
so what it's doing is staying on the current visual line number, which changes as the wrap happens. I think interaction should be possible, you would have to artificially slow down the wrapping to figure that out. When we're not jumping around that should be doable.

Reply:
update: rustfmt _does_ complain about spaces around `..`